docs/topics/auth/default.txt

```diff
@@ -354,18 +354,27 @@ If you have an authenticated user you want to attach to the current session
                 # Return an 'invalid login' error message.
                 ...
 
+    To choose the :ref:`authentication backend <authentication-backends>` used
```

Comment:
I think we should rearrange the following section a bit for clarity and brevity. I'd replace everything from here to the end of the `login()` section with a sub-heading or note titled "Selecting the authentication backend," with some text along these lines:

> When a user is logged in, Django persists both the user's ID and the authentication backend used to authenticate them in their session, so the same authentication backend can be used to fetch their details on a future request. The authentication backend to persist is selected as follows:

> 1. Use the value of the optional `backend` argument, if provided.

> 2. Use the value of the `user.backend` annotation, if present. This is for backwards-compatible pairing of `authenticate()` and `login()`; `authenticate()` sets the `user.backend` annotation on the `User` object it returns.

> 3. Use the only configured backend in `AUTHENTICATION_BACKENDS`, if there is only one.

> 4. Otherwise, raise an exception.

In cases 1 and 2, the value of the `backend` argument or `user.backend` should be a dotted import path string (like that found in `AUTHENTICATION_BACKENDS`), not the actual backend class.